api/importer/execute/resource.go

```diff
@@ -0,0 +1,90 @@
+package execute
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	apimodels "github.com/keptn/go-utils/pkg/api/models"
+)
+
+type KeptnResourcePusher struct {
+	endpointProvider KeptnConfigurationServiceEndpointProvider
+	doer             httpdoer
+}
+
+type resourceRequest struct {
+	Resources []*apimodels.Resource `json:"resources"`
+}
+
+func (p *KeptnResourcePusher) pushContent(url string, content io.ReadCloser, resourceURI string) (any,
+	error) {
+	defer content.Close()
+	contentBytes, err := io.ReadAll(content)
+	if err != nil {
+		return nil, fmt.Errorf("error reading resource content: %w", err)
+	}
+	encodedContent := base64.StdEncoding.EncodeToString(contentBytes)
+
+	marshaledBytes, _ := json.Marshal(
+		resourceRequest{
+			Resources: []*apimodels.Resource{
+				{
+					ResourceContent: encodedContent,
+					ResourceURI:     &resourceURI,
+				},
+			},
+		},
+	)
+
+	request, err := http.NewRequest(
+		http.MethodPost,
+		url,
+		bytes.NewReader(marshaledBytes),
+	)
+
+	if err != nil {
+		return nil, fmt.Errorf("error creating resource request: %w", err)
+	}
+
+	request.Header.Set("Content-Type", "application/json")
+	response, err := p.doer.Do(request)
+
+	if err != nil {
+		return nil, fmt.Errorf("error performing add resource request: %w", err)
+	}
+
+	responseBody := new(any)
+	if response.Header.Get("Content-Type") == "application/json" {
```

Comment:
Checking the header should be probably done by a `HasPrefix` in case the service adds `; charset=utf8` in the future

Reply:
Reworked the code for parsing the response body (I also aligned it a bit more to the processing we do for API processing)